marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mapping/entity-services/lib.sjs

```diff
@@ -300,14 +300,20 @@ function validatePropertyMapping(fullMapping, propertyName, sourcedFrom) {
     // As of trunk 10.0-20190916, mappings are being validated against entity schemas in the schema database.
     // This doesn't seem expected, as the validation will almost always fail.
     // Thus, this is not using es.mappingCompile, which does validation, and just invokes the transform instead.
-    let stylesheet = xdmp.xsltInvoke("/MarkLogic/entity-services/mapping-compile.xsl", xmlMapping);
-    xdmp.xsltEval(stylesheet, [], {staticCheck: true});
+    let stylesheet = fn.head(xdmp.xsltInvoke("/MarkLogic/entity-services/mapping-compile.xsl", xmlMapping));
+    xdmp.xsltEval(removeStandardFunction(stylesheet), [], {staticCheck: true});
   } catch (e) {
     // TODO Move this into a separate function for easier testing?
     return getErrorMessage(e);
   }
 }
 
+//This function removes import of "standard-library.xqy" from mapping xslt. Server Bug 54051
+function removeStandardFunction(stylesheet) {
```

Comment:
We need a unit test for this function to ensure it's behaving correctly. One risk here is that if the "to-delete" node doesn't exist, this will return an empty element. So we need at least two unit tests - a happy path where the node does exist, and then one where the node doesn't exist.

For the implementation, I recommend just doing this:

```
element {fn:node-name($node)} {
  $node/namespace::*,
  $node/attribute(),
  element()[fn:not(@href = '/MarkLogic/entity-services/standard-library.xqy')]
}
```

That will work regardless of whether the node to delete exists or not.

Reply:
Changed the implementation and added tests